pkg/termcorder/termcorder.go

```diff
@@ -0,0 +1,210 @@
+package termcorder
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/creack/pty"
+	"golang.org/x/term"
+)
+
+//Termcording represent a command/terminal recording
+type Termcording struct {
+	Config *Config
+	cmd    *exec.Cmd
+	out    io.Writer
+}
+
+//Config represents the command configuration
+type Config struct {
+	Filename      string
+	CmdName       string
+	CmdArgs       []string
+	QuietMode     bool
+	Append        bool
+	Interactive   bool
+	LogKeystrokes bool
+	PrintHelp     bool
+}
+
+const defaultFileName = "termcording"
+
+var cli *flag.FlagSet
+
+//Cmd sets the termcording's cmd.
+func Cmd(c *exec.Cmd) func(*Termcording) error {
+	return func(tc *Termcording) error {
+		tc.cmd = c
+		return nil
+	}
+}
+
+//Output sets the termcording's output.
+func Output(w io.Writer) func(*Termcording) error {
+	return func(tc *Termcording) error {
+		tc.out = w
+		return nil
+	}
+}
+
+//NewTermcording returns a pointer to a new variable of type Termcording given a config
+//variable and (functional) options.
+func NewTermcording(c *Config, options ...func(*Termcording) error) (*Termcording, error) {
+	tc := &Termcording{
+		Config: c,
+	}
+	for _, option := range options {
+		err := option(tc)
+		if err != nil {
+			return &Termcording{}, err
```

Comment:
The zero value for pointer types is `nil`, so if we want to return the zero value in the error case, that would make sense, wouldn't it? We usually only return an empty struct when we return the struct type itself, because structs can't be nil.